core/trino-main/src/test/java/io/trino/operator/output/TestSkewedPartitionRebalancer.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.operator.output;
+
+import com.google.common.collect.ImmutableList;
+import io.airlift.units.DataSize;
+import io.trino.SequencePageBuilder;
+import io.trino.operator.PartitionFunction;
+import io.trino.spi.Page;
+import it.unimi.dsi.fastutil.ints.IntArrayList;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static io.airlift.units.DataSize.Unit.MEGABYTE;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestSkewedPartitionRebalancer
+{
+    @Test
+    public void testRebalanceWithSkewness()
```

Comment:
https://github.com/trinodb/trino/commit/106ee5d7fafe5d241985b0e732ee8a106b14daf8 introduced more tests. Do you think these 3 tests are sufficient?